lib/Index/Index.cpp

```diff
@@ -1378,20 +1383,37 @@ bool IndexSwiftASTWalker::reportInheritedTypeRefs(ArrayRef<InheritedEntry> Inher
   return true;
 }
 
-bool IndexSwiftASTWalker::reportRelatedTypeRef(const TypeLoc &Ty, SymbolRoleSet Relations, Decl *Related) {
-  if (auto *declRefTR = dyn_cast_or_null<DeclRefTypeRepr>(Ty.getTypeRepr())) {
-    SourceLoc IdLoc = declRefTR->getLoc();
+bool IndexSwiftASTWalker::reportRelatedTypeRef(const TypeLoc &Ty, SymbolRoleSet Relations,
+                                               Decl *Related, bool Implicit, const Optional<SourceLoc> Loc) {
```

Comment:
I actually found the need for const when I accidentally typed `Loc = IdLoc` instead of the other way around and it compiled, w/ const it doesn't anymore. Is that a reasonable use of const? Will add docs!

Reply:
I suppose it's one use, I don't often see it being used like that though. Ie. it being on a value rather than a ref was fairly surprising to me.

Reply:
Sounds good, I removed it and added docs!

Reply:
> I actually found the need for const when I accidentally typed Loc = IdLoc instead of the other way around and it compiled, w/ const it doesn't anymore. Is that a reasonable use of const?

Just a side-note, it'd be great if all value parameters in C++ were "const" (like in Swift) or LLVM/Swift coding conventions mandated making them `const` everywhere, but since that is not the case, making one value parameter `const` does seem surprising and "out-of-place" in this codebase.